ENH - update README (#549)
benchopt/benchopt: README.rst

.. raw:: html

    <embed>
        <div class="container" align="center">
         <img src="https://raw.githubusercontent.com/benchopt/communication_materials/main/posters/images/logo_benchopt.png" width="350em"/>
        </div>
        <em>—Making your optimization benchmarks simple and open—</em>
        <h1></h1>
    </embed>

|Test Status| |codecov| |Python 3.6+| |install-per-months| |discord| 

``Benchopt`` is a benchmarking suite for optimization algorithms.
It is built for simplicity, transparency, and reproducibility.
It is implemented in Python but can run algorithms written in **many programming languages**.

So far, ``benchopt`` has been tested with `Python <https://www.python.org/>`_,
`R <https://www.r-project.org/>`_, `Julia <https://julialang.org/>`_
and `C/C++ <https://isocpp.org/>`_ (compiled binaries with a command line interface).

> Programs available via `conda <https://docs.conda.io/en/latest/>`_ should be compatible as well. See an `example of usage <https://benchopt.github.io/auto_examples/plot_run_benchmark_python_R.html>`_ with ``R``.

Reviewer: ```suggestion
Programs available via `conda <https://docs.conda.io/en/latest/>`_ should be compatible as well.
See for instance an `example of usage <https://benchopt.github.io/auto_examples/plot_run_benchmark_python_R.html>`_ with ``R``.
```

Install
-------

It is recommended to use ``benchopt`` within a ``conda`` environment to fully-benefit
from ``benchopt`` Command Line Interface (CLI).  

To install ``benchopt``, start by creating a new ``conda`` environment and then activate it

.. code-block::

    conda create -n benchopt python
    conda activate benchopt

> Finally run the following command to install the **latest release** of ``benchopt``

Reviewer: ```suggestion
Then run the following command to install the **latest release** of ``benchopt``
```

.. code-block::

    pip install -U benchopt

> It is possible to use the **latest development version**. To do so, run instead

Reviewer: ```suggestion
It is also possible to use the **latest development version**. To do so, run instead
```

.. code-block::

    pip install -U -i https://test.pypi.org/simple/benchopt

Getting started 
---------------

After installing ``benchopt``, you can

> - replicate/modify and existing benchmark

Reviewer: ```suggestion
- replicate/modify an existing benchmark
```

- create your own benchmark

Using an existing benchmark
^^^^^^^^^^^^^^^^^^^^^^^^^^^

> Replicating and exiting benchmark is simple.

Reviewer: ```suggestion
Replicating an existing benchmark is simple.
```

> Here is how to do it for the `L2-logistic Regression benchmark <https://github.com/benchopt/benchmark_logreg_l2>`_.

Reviewer: ```suggestion
Here is how to do so for the `L2-logistic Regression benchmark <https://github.com/benchopt/benchmark_logreg_l2>`_.
```

1. Clone the benchmark repository and ``cd`` to it

.. code-block:: bash

   git clone https://github.com/benchopt/benchmark_logreg_l2
   cd benchmark_logreg_l2

> 2. Install the desired solvers

Reviewer: ```suggestion
2. Install the desired solvers automatically with ``benchopt``
```

.. code-block:: bash

   benchopt install . -s lightning -s sklearn

3. Run the benchmark to get the figure below

.. code-block:: bash

   benchopt run . --config ./example_config.yml

.. figure:: https://benchopt.github.io/_images/sphx_glr_plot_run_benchmark_001.png
   :target: how.html
   :align: center
   :scale: 40%

These steps illustrate how to reproduce the `L2-logistic Regression benchmark <https://github.com/benchopt/benchmark_logreg_l2>`_.
Find the complete list of the `Available benchmarks`_.

> Also, refer to the `documentation <https://benchopt.github.io/>`_ to learn more about ``benchopt`` CLI and its features.

Reviewer: ```suggestion
Also, refer to the `documentation <https://benchopt.github.io/>`_ to learn more about ``benchopt`` CLI and its features.
You can also easily extend this benchmark by adding a dataset, solver or metric.
Learn how to do so in the different sections of this tutorial: `Write a benchmark <https://benchopt.github.io/how.html>`_.
```

Creating a benchmark
^^^^^^^^^^^^^^^^^^^^

> The section `Write a benchmark <https://benchopt.github.io/how.html>`_ of the documentation provides a nice tutorial

Reviewer: ```suggestion
The section `Write a benchmark <https://benchopt.github.io/how.html>`_ of the documentation provides a tutorial
```

for creating a benchmark. The ``benchopt`` community also maintains 

> a `template benchmark <https://github.com/benchopt/template_benchmark>`_ to quickly and easily write a benchmark.

Reviewer: ```suggestion
a `template benchmark <https://github.com/benchopt/template_benchmark>`_ to quickly and easily start a new benchmark.
```

Finding helps
-------------

Join ``benchopt`` `discord server <https://discord.gg/Rxwjh9ApR9>`_ and get in touch with the community!
Feel free to drop us a message to get help with running/constructing benchmarks 
or (why not) discuss new features to be added and future development directions that ``benchopt`` should take.

Citing Benchopt
---------------

> ``Benchopt`` is a continuous endeavor to make reproducible and transparent optimization benchmarks.

Reviewer: ```suggestion
``Benchopt`` is a continuous effort to make reproducible and transparent optimization benchmarks.
```
